index.d.ts

```diff
@@ -0,0 +1,111 @@
+/// <reference types="react" />
+
+import {ClassicComponentClass} from 'react';
```

Comment:
It does help editors indeed but it's also required by the typescript transpiler.

Without a type definition, the typescript transpiler will throw error such as these :
> ERROR in ./app/components/my-component.tsx
> (136,17): error TS2604: JSX element type 'Waypoint' does not have any construct or call signatures.
> 
> ERROR in ./app/components/my-component.tsx
> (10,28): error TS2307: Cannot find module 'react-waypoint'.
Depending on your configuration, it might break your build. At the very least it clutters the output with meaningless errors.

As stated in the publishing guide, it can be bundled with the module or live a in a separate one (most likely @types/react-waypoint). The former makes it simpler to use since you don't have to look for a definition file manually.

You can ask the transpiler to generate such file but for that you'll obviously have to code in typescript which I'm guessing isn't what you're aiming for :)